### PR TITLE
Document the CAQL endpoint

### LINKED ISSUE #30: Document CAQL endpoint
Currently we only have generic lua extensions documented. We should add some description how to access CAQL in particular.

It might make sense to create a new API endpoint `/caql` for CAQL that proxies to `/extension/lua/caql_v1.lua`.

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
Resolves #30

### CODE REVIEW

File: api/extension-caql.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+* `query` : The query to execute.
+
+* `start` : The start time from which to pull data, represented in seconds since the epoch. This value is inclusive (data for the given start time will be pulled).
+
+* `end` : The end time up to which data is pulled, represented in seconds since the epoch. This value is exclusive (data up to, but not including, the given end time will be pulled).
+
+* `period` : The period, in seconds, for which to get data rollups.
+
+* `_timeout` : Spefify a timeout for CAQL processing in seconds. Optional. Default = 4.5.
```
Comment (esproul) [inline api/extension-caql.md]: "Specify" (typo)

File: api/extension-caql.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+`/extension/lua/caql_v1`
+
+### Method
+
+GET
+
+### Inputs
+
+Parameters can be submitted as query string parameters or as JSON payload.
```
Comment (esproul) [inline api/extension-caql.md]: Might want to mention that, if provided via query string parameter, the query needs to be URL-encoded.

File: api/extension-caql.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+```
+echo '{
+  "q":"12+3",
+  "start":1474275000,
+  "end":1474275240,
+  "period":60
+}' | curl http://127.0.0.1:8112/extension/lua/caql_v1 --data @-
+```
+
```
Comment (esproul) [inline api/extension-caql.md]: Might be helpful if we could see the same query given using query-string parameters.

Comment (esproul) [reply api/extension-caql.md]: Tiny nit: if someone copy/pastes the query-string example, the URL will need to be single-quoted to protect the shell from `?` and `&`.

Comment (HeinrichHartmann) [reply api/extension-caql.md]: That is true.

**Comment by HeinrichHartmann** [comment]
merging now.

**Pull request merged**